### PR TITLE
Updated readme to include neovim snippet

### PR DESCRIPTION
**Pull request by Rexcrazy804**
added a snippet for correctly setting up nvim-lspconfig and re-structured the relevant portions of the document to make the modification cohesive. 

[ it took me a fair bit of time to set this up without Folke's neoconf, so for anyone who may need it, this will save them quite a bit of time]

### CODE REVIEW

**Comment by Rexcrazy804** [comment]
I think I can contribute to the neovim section for #107 
Should I make neovim a heading and start from there? I'll leave the vscode as is since I am not so familiar with it.

File: README.md [hunk 1]
```diff
@@ -46,14 +46,28 @@ You can set the `rust-analyzer.rustfmt.overrideCommand` setting.
 ```json
   "rust-analyzer.rustfmt.overrideCommand": ["leptosfmt", "--stdin", "--rustfmt"]
 ```
+> Note: For VSCode users, I recommend to use workpsace settings (CMD + shift + p -> Open workspace settings), so that you can only configure `leptosfmt` for workpsaces that are using leptos. For Neovim users,
+
+Alternatively, if you are using [nvim-lspconfg](https://github.com/neovim/nvim-lspconfig), you may append the following to your `.setup{}` table:
```
Comment (bram209) [inline README.md:51]: I am pretty sure that `neoconf.nvim` still utilises `nvim-lspconfig` under the hood.
```suggestion
Alternatively, you can configure [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) directly, you may append the following to your `.setup{}` table:
```

Comment (bram209) [reply README.md:51]: we could also point them to the docs of `nvim-lspconfig`: https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#rust_analyzer

Comment (Rexcrazy804) [reply README.md:51]: roger, I'll make these changes

**Comment by Rexcrazy804** [comment]
do you rekon I update this branch? (since its showing that it is out-of-date with the base branch)

**Comment by bram209** [comment]
Hi, this one slipped through, sorry for the late reply.

Considering there is now a new way to configure `rust-analyzer` (with a `rust-analyzer.toml` file), I rewrote the documentation slightly. While doing this, I incorporated your documentation about neovim as well.

You can see the changes here: https://github.com/bram209/leptosfmt/commit/85ad11bae8188e73f20e45efbdaf6ce77564f00d

Thank you for your contribution!

**Pull request closed without merging**